all.go

```diff
@@ -134,34 +134,67 @@ func (n *Node) all(tx *bolt.Tx, info *modelInfo, ref *reflect.Value, rtyp, typ r
 
 	if bucket != nil {
 		c := bucket.Cursor()
-		for k, v := c.First(); k != nil; k, v = c.Next() {
-			if v == nil {
-				continue
+		if opts.Reverse {
+			for k, v := c.Last(); k != nil; k, v = c.Prev() {
```

Comment:
You need to extract a method or a func to avoid the mighty copy-paste (to keep it DRY ...)